lib/diaspora_federation/schemas/federation_entities.json

```diff
@@ -0,0 +1,366 @@
+                    { "$ref": "#/definitions/location" },
+                    { "type": "null" }
+                  ]
+                },
+
+                "poll": {
+                  "oneOf": [
+                    { "$ref": "#/definitions/poll" },
+                    { "type": "null" }
+                  ]
+                },
+
+                "photos": {
+                  "type": ["array", "null"],
+                  "items": { "$ref": "#/definitions/photo" }
+                }
+              },
+
+              "required": ["text", "location", "poll", "photos"]
+            }
+          }
+        }
+      ]
+    },
+
+    "reshare": {
+      "allOf": [
+        {"$ref": "#/definitions/post"},
+        {
+          "type": "object",
+          "properties": {
+            "entity_type": {
+              "type": "string",
+              "pattern": "^reshare$"
+            },
+
+            "entity_data": {
+              "type": "object",
+              "properties": {
+                "root_author": {"type": "string"},
+                "root_guid": {"type": "string"}
+              },
+
+              "required": ["root_author", "root_guid"]
+            }
+          }
+        }
+      ]
+    },
+
+    "profile": {
+      "type": "object",
+      "properties": {
+        "entity_type": {
+          "type": "string",
+          "pattern": "^profile$"
+        },
+        "entity_data": {
+          "type": "object",
+          "properties": {
+            "first_name": { "type": ["string", "null"] },
+            "last_name": { "type": ["string", "null"] },
+            "gender": { "type": ["string", "null"] },
+            "bio": { "type": ["string", "null"] },
+            "birthday": { "type": ["string", "null"] },
+            "location": { "type": ["string", "null"] },
+            "image_url": { "type": ["string", "null"] },
+            "author": { "type": "string" }
+          },
+          "required": [
+            "first_name", "last_name", "gender", "bio", "birthday", "location", "image_url", "author"
+          ]
+        }
+      },
+      "required" :["entity_data", "entity_type"]
+    },
+
+    "photo": {
+      "type": "object",
+      "properties": {
+        "entity_type": {
+          "type": "string",
+          "pattern": "^photo$"
+        },
+        "entity_data": {
+          "type": "object",
+          "properties": {
+            "guid": {"type": "string"},
+            "public": {"type": "boolean"},
+            "created_at": {"type": "string"},
+            "remote_photo_path": {"type": "string"},
+            "remote_photo_name": {"type": "string"},
+            "text": {"type": ["null", "string"]},
+            "status_message_guid": {"type": "string"},
+            "width": {"type": "number"},
+            "height": {"type": "number"}
+          },
+          "required": [
+            "guid", "remote_photo_path", "remote_photo_name", "status_message_guid", "width", "height"
```

Comment:
`author` is missing and required, `public` and `created_at` are also required, `status_message_guid` is optional.